src/logger/loggerTaskEx.c

```diff
@@ -243,8 +247,7 @@ void loggerTaskEx(void *params)
                 }
 
                 /* send the sample on to the telemetry task(s) */
-                if (sampledRate >= telemetrySampleRate ||
-                    currentTicks % telemetrySampleRate == 0)
+                if (!isHigherSampleRate(sampledRate, telemetrySampleRate))
```

Comment:
So my concern is you removed the modulus operator-  IIRC it's there to prevent a user from selecting a combination of sample rates in their logger config that would prevent telemetry from broadcasting.  My sense it may need more testing of different sample rate combinations and ensuring telemetry broadcast works.. That's the regression risk I'm seeing, and why I think it should be moved to 2.8.6.